projects/opendr_ws/src/perception/scripts/video_activity_recognition.py

```diff
@@ -101,23 +120,27 @@ def callback(self, data):
         :param data: input message
         :type data: sensor_msgs.msg.Image
         """
-        image = self.bridge.from_ros_image(data)
+        image = self.bridge.from_ros_image(data, encoding="rgb8")
         if image is None:
             return
 
-        x = self.preprocess(image.numpy())
+        x = self.preprocess(image.convert("channel_first", "rgb"))
 
         result = self.learner.infer(x)
```

Comment:
When running with a webcam input, using all the default node arguments, this line produces the following error:

`RuntimeError: Given input size: (432x6x7). Calculated output size: (432x0x1). Output size is too small`

Let me know if it would be helpful to post the whole stack trace.

Reply:
Thanks for checking @tsampazk. Frankly, we're a bit challenged wrt. ROS1 testing on our side, as our only Ubuntu laptop is 22.04, which I haven't been able to install ROS1 on. Checks have made on a Ubuntu 18 server, which is locked away in a basement without a cam 😅 
The above error seems to be due to wrong ordering of input dimensions. I hope the "channels_first: will fix this - thanks for catching the error

Reply:
No problem! Unfortunately, with `channels_first` on the preprocess method i get this error. I messed around and tried `channels_last`, but then the node just gets killed after a while, due to some silent error during the preprocessing step, which seems expected because the preprocessing functions seem to expect a channels first image.

Before the preprocessing step the image shape is `(3, 480, 640)` and after it becomes (3, 168, 224) with channels first obviously. 

If you have any other ideas that i can try, let me know.

Reply:
Thanks a lot of supporting the debugging here :-) We want minimum 224x224 as input, so I think the resizing is the issue here. I've pushed a potential fix 🤞

Reply:
Alright, the new changes fix the issue! There is one more minor error that's easily fixable. In line 132, it should either be:
`category.confidence = float(category.confidence.max())`
or 
`category.confidence = float(max(category.confidence))`
because `category.confidence` is a tensor and max is taken twice otherwise.

A minor note is that i get this warning:

```opendr/projects/opendr_ws/src/perception/scripts/video_activity_recognition.py:182: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  return Image(frame, dtype=np.float)```

which is fixable by making line 182 `return Image(frame, dtype=float)`.

Reply:
After this the node runs properly, publishing results as expected.

Reply:
Great, you saved the day :-) Thank you @tsampazk

Reply:
Happy to help :)